common/courses/states.ts

```diff
@@ -32,18 +74,80 @@ export async function canPublish(subcourse: Subcourse): Promise<Decision> {
     return { allowed: true };
 }
 
-export async function subcourseOver(subcourse: Subcourse) {
-    const lastLecture = await prisma.lecture.findFirst({
-        where: { subcourseId: subcourse.id },
-        orderBy: { start: 'desc' },
-    });
+export async function publishSubcourse(subcourse: Subcourse) {
+    const can = await canPublish(subcourse);
+    if (!can.allowed) {
+        throw new Error(`Cannot Publish Subcourse(${subcourse.id}), reason: ${can.reason}`);
+    }
 
-    if (!lastLecture) {
-        return false;
+    await prisma.subcourse.update({ data: { published: true, publishedAt: new Date() }, where: { id: subcourse.id } });
+    logger.info(`Subcourse (${subcourse.id}) was published`);
+
+    const course = await getCourse(subcourse.courseId);
+    await sendPupilCourseSuggestion(course, subcourse, 'instructor_subcourse_published');
```

Comment:
Maybe we should add logs to show that the suggestion is executed.

Reply:
True, I have the feeling that we will have to rework the course promotion feature once that spams our users anyways, so :)
Did some minor cleanup and added logging